transmission_interface/src/transmission_parser.cpp

```diff
@@ -229,7 +229,7 @@ bool TransmissionParser::parseActuators(TiXmlElement *trans_it, std::vector<Actu
     }
     if (actuator.hardware_interfaces_.empty())
     {
-      ROS_DEBUG_STREAM_NAMED("parser","No valid hardware interface element found in actuator '"
+      ROS_DEBUG_STREAM_NAMED("parser","Optional: No valid hardware interface element found in actuator '"
```

Comment:
IMHO, it'd be more clear to add a counter in the previous `for` loop and provide two different messages depending on whether the counter is `== 0` or `> 0`. The current message here looks fine for the case when `> 0`, maybe adding another sentence saying that the actuator is added anyway. When `== 0` the message could just be something like `Actuator <ACTUATOR> with no hardware interfaces found.`

BTW, I think the `if(!hw_iface_it) {continue;}` block above is useless, because the `for` loop already has `hw_iface_it` on the condition.